elections/views.py

```diff
@@ -160,20 +167,43 @@ def get_rank(vote):
 def results(request):
     authenticate(request, VALID_FACTORS)
     #add a condition to check if the election is closed and exists
-    quorum = 350
 
-    # Ballot.objects.filter(quorum=True)
-    try:
-        elections = Election.objects.all()
-        for election in elections:
+    #get the elections that are finishes & that we can get results for
+    #relection is a list of valid elections, 
+    #nonvanity is a boolean; if there's at least one election that's not vanity (requires quorum)
+    nonvanity = False
+    elections = Election.get_closed()
+    relections = []
+    participation = 0.0
+
+    for election in elections:
+        #check to see if non-vanity elections meet quorum
+        if not election.vanity:
+            b = Ballot.objects.filter(election=election).exclude(votes="noquorum")
+            #set participation to the max of all non-vanity elections
+            if (len(b)/float(SB_SIZE))*100 > participation:
+                participation = (len(b)/float(SB_SIZE))*100
+            if len(b) >= QUORUM:
+                relections.append(election)
+                nonvanity = True
+                winners = calculateSTV(election)
+                for candidate in winners:
+                    election.results.add(candidate)
+                print election.results.all()
+        #we don't need to check for quorum
+        else:
+            relections.append(election)
             winners = calculateSTV(election)
```

Comment:
We still need to have the logic save the winners and only run the algorithm when that is blank. This means we need to add a new field to the `Election` model.